js/views/site.es6.js

```diff
@@ -23,12 +23,28 @@ function Site (ops) {
             thisModel.domain = backgroundPage.utils.extractHostFromURL(tab.url);
             thisModel.tabId = tab.id;
             thisModel.setSiteObj();
+
+                                      
+            if (thisModel.disabled) {   // determined in setSiteObj()
+                thisView.setDisabled();
+            }
+
             thisModel.updateTrackerCount();
             thisModel.setHttpsMessage();
             thisView.rerender(); // our custom rerender below
         }
+        else {
+            console.debug('Site view: no tab');
+        }
     });
 
+    // edge case, should not happen
+    // '-' is the domain default in the pages/trackers.es6.js call
+    if (this.domain === '-') {
+        this.model.disabled = true;
+        this.setDisabled();
```

Comment:
this is split because it's used differently elsewhere.
(see line 29 above, in that case `disabled` is set separately in the model)
could send the value as an argument, and if the argument is present, set it, otherwise just set the class on body.